lib/api/v1/marketing/lists.rb

```diff
@@ -0,0 +1,670 @@
+                # bugger_add('hbx_enrollment...') # bugger
+                # bugger_add(pp hbx_enrollment) # bugger
+                # if ! hbx_enrollment.nil? then # add our amount due field...
+                #     tmp[:total_premium] = hbx_enrollment.total_premium.to_s # some kind of on-demand caluation method?
+                #     bugger_add('hbx_enrollment.total_premium: ' + tmp[:total_premium]) # bugger
+                # end
+
+                plans.push(tmp)
+            end
+        end
+    end
+
+    # 
+    # get_enrolled_members()
+    #   
+    def get_enrolled_members (enr, family)
+        bugger_add('get_enrolled_members()...') # bugger
+        persons = []
+        person_ids = []
+        family_map = {}
+
+        family.each do |fm|
+            family_map[fm[:_id]] = fm[:person_id]
+        end
+
+        enr[:hbx_enrollment_members].each do |m|
+            person_ids.push(family_map[m[:applicant_id]])
+        end
+
+        p = CollPeople.only(:first_name).where("_id": { '$in': person_ids })
+
+        p.each do |person|
+            persons.push(person[:first_name])
+        end
+
+        return persons
+    end
+
+    # 
+    # get_person_consumer()
+    #   
+    def get_person_consumer (id)
+        bugger_add('get_person(' + id + ')...') # bugger
+
+        p = CollPeople.only(:first_name, :last_name, :emails, :consumer_role)
+           .where("_id": id)
+           .where("consumer_role.is_active": true)
+           .where(:emails.nin => [nil, []])
+
+        return p.length > 0 ? p[0] : false 
+    end
+
+    # 
+    # ck_api_key()
+    #   
+    def ck_api_key (user, key, type)
+        ok = false
+        user_ok = @caller.params['user'] == user ? true : false
+        bugger_add('ck_api_key()...') # bugger
+        if (@caller.params['user'] == user)
+            bugger_add(' user: ok') # bugger
+            if (type == 'plain' && @caller.params['key'] == key)
+                ok = true
+                bugger_add(' key (plain): ok') # bugger
+            elsif (type == 'hashed' && ck_hash(@caller.params['key'], key))
+                ok = true
+                bugger_add(' key (hashed): ok') # bugger
+            else
+                bugger_add(' key: fail') # bugger
+            end
+        else
+            bugger_add(' user: fail') # bugger
+        end
+
+        return ok
+    end
+
+    # 
+    # ck_hash()
+    #   
+    def ck_hash (plain, hashed)
+        ok = false
+        require 'bcrypt'
+        hashed_pass = BCrypt::Password.new(hashed)
+        ok = true if hashed_pass == plain
+
+        return ok
+    end
+
+    # 
+    # auth()
+    #   
+    def auth
+        ok = false
+        ip_ok = false
+
+        bugger_add('auth()...') # bugger
+        bugger_add('found key: allowed_ips') if @app_config['allowed_ips'] # bugger
+        bugger_add({'@app_config' => @app_config}) # bugger
+
+        if ! @app_config['allowed_ips'] 
+            ip_ok = true
```

Comment:
Can we take out this option? I realize we may not know the incoming IPs for preprod and prod yet. But we do, ultimately, want to restrict it to specific incoming IPs, and I'm a little worried by the idea that a config change could open this API up, given the massive amount of PII it exposes. I'd rather not even have an "everything visible to the public" codepath in here.